master/internal/db/postgres_validations.go

```diff
@@ -0,0 +1,116 @@
+package db
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	_ "github.com/golang-migrate/migrate/source/file" // Load migrations from files.
+	_ "github.com/jackc/pgx/v4/stdlib"                // Import Postgres driver.
+	"github.com/pkg/errors"
+
+	"github.com/determined-ai/determined/master/pkg/model"
+)
+
+// AddValidation adds the validation to the database and sets its ID.
+func (db *PgDB) AddValidation(validation *model.Validation) error {
+	if !validation.IsNew() {
```

Comment:
Same comments here as for checkpoints. Removing the constraint that there isn't a step associated with the validation fits with the rest of this PR and may now be an OK state for the backend but I'm not sure how the webui will handle. Have you looked into this?

Reply:
Same.